adapta/utils/README.md

```diff
@@ -107,4 +107,44 @@ with memory_limit(memory_limit_percentage=0.2):
     # Do some stuff here
     # if the provided limit is exceeded, MemoryError will be raised and can be caught by client code
     pass
-```
\ No newline at end of file
+```
+
+### runtime_metrics:
+
+Decorator for logging method execution time and reporting function execution time to metric provider (e.g. DataDog).  
+
+Notes:
+* When using the decorator the wrapped function must take logger and metrics_provider as keyword arguments.
+  * It is recommended to have decorated methods use `**kwargs` if logger and metric providers are not explicitly defined.
+* Logger sends logs with `loglevel.DEBUG`.
+
+Parameters:
+* `metric_name`: Type of method being wrapped. This will be used as a metric_name by the metric provider.
+* `tag_function_name` (optional): If True function name will be added as additional metric_tag. Defaults to False.
+* `metric_tags` (optional): Dictionary containing extra tags used by the metric provider.
+
+Example Usage:
+
+```python
+import time
+from adapta.metrics.providers.datadog_provider import DatadogMetricsProvider
+from adapta.logs import SemanticLogger
+from adapta.logs.models import LogLevel
+from adapta.utils import run_time_metrics
+
+
+@run_time_metrics(metric_name='example')
+def example_function(logger=None, metrics_provider=None, **kwargs):
```

Comment:
I'd say show two possible sigs - one with arg1, arg2 and kwargs, and one with arg1, logger, metrics_provider, kwargs